qctrlopencontrols/dynamic_decoupling_sequences/driven_controls.py

```diff
@@ -216,22 +223,18 @@ def convert_dds_to_driven_control(
         operations.shape[1], 2))   # pylint: disable=unsubscriptable-object
 
     for op_idx in range(operations.shape[1]):   # pylint: disable=unsubscriptable-object
+        # Pulses that cause no rotations can have 0 duration
+        half_pulse_duration  = 0.
 
-        if operations[3, op_idx] == 0: #no z_rotations
-            if not np.isclose(operations[1, op_idx], 0.):
-                half_pulse_duration = 0.5 * operations[1, op_idx] / maximum_rabi_rate
-            else:
-                half_pulse_duration = 0.5 * operations[2, op_idx] / maximum_rabi_rate
-
-            pulse_start_ends[op_idx, 0] = pulse_mid_points[op_idx] - half_pulse_duration
-
-            pulse_start_ends[op_idx, 1] = pulse_mid_points[op_idx] + half_pulse_duration
-        else:
-            half_pulse_duration = 0.5 * np.abs(operations[3, op_idx]) / maximum_detuning_rate
-
-            pulse_start_ends[op_idx, 0] = pulse_mid_points[op_idx] - half_pulse_duration
+        if not np.isclose(operations[1, op_idx], 0.): # Rabi rotation
+            half_pulse_duration = 0.5 * max(operations[1, op_idx] / maximum_rabi_rate,
+                                            minimum_segment_duration)
+        elif not np.isclose(operations[3, op_idx], 0.): # Detuning rotation
+            half_pulse_duration = 0.5 * max(np.abs(operations[3, op_idx]) / maximum_detuning_rate,
+                                            minimum_segment_duration)
 
-            pulse_start_ends[op_idx, 1] = pulse_mid_points[op_idx] + half_pulse_duration
+        pulse_start_ends[op_idx, 0] = pulse_mid_points[op_idx] - half_pulse_duration
+        pulse_start_ends[op_idx, 1] = pulse_mid_points[op_idx] + half_pulse_duration
```

Comment:
Yes, it may, and yes, this is one of the reason for the extra test below. The other reason is that there's no guarantee that the spacings between the pulses will be sufficient to satisfy a minimum_segment_duration even before that shift